var/spack/repos/builtin/packages/paraview/package.py

```diff
@@ -363,9 +366,42 @@ def nvariant_bool(feature):
         else:
             cmake_args.append('-DVTKm_ENABLE_CUDA:BOOL=%s' %
                               variant_bool('+cuda'))
-        if spec.satisfies('+cuda') and not spec.satisfies('cuda_arch=native'):
-            cmake_args.append('-DVTKm_CUDA_Architecture=%s' %
-                              spec.variants['cuda_arch'].value)
+
+        # VTK-m expects cuda_arch to be the arch name vs. the arch version.
+        if spec.satisfies('+cuda'):
+            supported_cuda_archs = {
+
+                # VTK-m and transitively ParaView does not support Tesla Arch
+                '20': 'fermi',
+                '21': 'fermi',
+                '30': 'kepler',
+                '32': 'kepler',
+                '35': 'kepler',
+                '37': 'kepler',
+                '50': 'maxwel',
+                '52': 'maxwel',
+                '53': 'maxwel',
+                '60': 'pascal',
+                '61': 'pascal',
+                '62': 'pascal',
+                '70': 'volta',
+                '72': 'volta',
+                '75': 'turing',
+                '80': 'ampere',
+                '86': 'ampere',
+            }
+
+            cuda_arch_value = 'native'
+            requested_arch = spec.variants['cuda_arch'].value
+
+            # ParaView/VTK-m only accepts one arch, default to first element
+            if requested_arch[0] != 'none':
+                try:
+                    cuda_arch_value = supported_cuda_archs[requested_arch[0]]
+                except KeyError:
+                    raise InstallError("Incompatible cuda_arch=" + requested_arch[0])
```

Comment:
Just an idea, but did you try doing something like:
```python
for _arch, _other_arch in itertools.permutations(CudaPackage.cuda_arch_values, 2):
    conflicts(
        'cuda_arch={0}'.format(_arch), 
        when='cuda_arch={0}'.format(_other_arch), 
        msg='Paraview accepta only one architecture value.';
    )
```
? I'm not sure how this formulation would affect concretization time, given the number of conflicts, but in principle it permits to delete these checks at runtime and move them at concretization time.

Reply:
@alalazo find this resolved at the latest push, however, note that when the conflict happens the error _msg_ does not print out